cmake/kokkos_compiler_id.cmake

```
@@ -94,6 +94,11 @@ if(Kokkos_ENABLE_HIP)
 
   string(REGEX MATCH "[0-9]+\\.[0-9]+\\.[0-9]+" TEMP_CXX_COMPILER_VERSION ${INTERNAL_COMPILER_VERSION_ONE_LINE})
   set(KOKKOS_CXX_COMPILER_VERSION ${TEMP_CXX_COMPILER_VERSION})
+
+  #FIXME HIP
+  if(KOKKOS_CXX_COMPILER_VERSION MATCHES "7\\.1")
```

```suggestion
  if(KOKKOS_CXX_COMPILER_ID STREQUAL HIPCC AND KOKKOS_CXX_COMPILER_VERSION VERSION_EQUAL 7.1)
```
How would `amdclang` behave?

---

The same way as `hipcc`. They are the same compiler with different flags.

---

yeah, I checked, for 7.1 our detection gave the same version number for `hipcc` and `amdclang`

---

```suggestion
  if(KOKKOS_CXX_COMPILER_VERSION VERSION_EQUAL 7.1)
```

---

> yeah, I checked, for 7.1 our detection gave the same version number for `hipcc` and `amdclang`

On Frontier, I'm seeing
```
$ cmake -DCMAKE_CXX_COMPILER=amdclang++ -DKokkos_ENABLE_TESTS=ON -DKokkos_ENABLE_HIP=ON -DCMAKE_CXX_STANDARD=20 ..
-- Setting default Kokkos CXX standard to 20
-- The CXX compiler identification is Clang 18.0.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /opt/rocm-6.2.4/bin/amdclang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Setting build type to 'RelWithDebInfo' as none was specified.
-- Kokkos version: 5.0.99
-- The project name is: Kokkos
-- Using bundled GoogleTest version
CMake Warning at cmake/kokkos_check_env.cmake:13 (message):
  CRAYPE_LINK_TYPE is not set.  Linking is likely to fail unless this is set
  to 'dynamic'
Call Stack (most recent call first):
  CMakeLists.txt:205 (include)


-- Configured git information in /ccs/home/darndt/kokkos/build/generated/Kokkos_Version_Info.cpp
-- Kokkos is configured for CMake languages CXX compilation (using Clang version 18.0.0)
-- SERIAL backend is being turned on to ensure there is at least one Host space. To change this, you must enable another host execution space and configure with -DKokkos_ENABLE_SERIAL=OFF.
-- Using -std=gnu++20 for C++20 extensions as feature
-- Setting Kokkos_ARCH_VEGA90A=ON
-- Built-in Execution Spaces:
--     Device Parallel: Kokkos::HIP
--     Host Parallel: NoTypeDefined
--       Host Serial: SERIAL
-- 
-- Architectures:
--  VEGA90A
-- Found TPLROCM: /opt/rocm-6.2.4/lib/libamdhip64.so  
-- Found Threads: TRUE  
-- Performing Test HIP_CLANG_SUPPORTS_PARALLEL_JOBS
-- Performing Test HIP_CLANG_SUPPORTS_PARALLEL_JOBS - Success
-- Using bundled desul_atomics copy (desul/desul@79f928075837ffb5d302aae188e0ec7b7a79ae94)
-- Performing Test KOKKOS_LINK_OPTIONS_CHECK
-- Performing Test KOKKOS_LINK_OPTIONS_CHECK - Success
-- Performing Test KOKKOS_COMPILE_OPTIONS_CHECK
-- Performing Test KOKKOS_COMPILE_OPTIONS_CHECK - Success
-- Using bundled mdspan copy (kokkos/mdspan@537053f83150f1b6f7528c1d961866d8628abc2e)
-- Found Python3: /usr/bin/python3.12 (found version "3.12.9") found components: Interpreter 
-- Sources TestHIP.cpp
-- Kokkos Backends: SERIAL;HIP
-- Configuring done (5.8s)
-- Generating done (2.6s)
-- Build files have been written to: /ccs/home/darndt/kokkos/build
```
in particular
> -- Kokkos is configured for CMake languages CXX compilation (using Clang version 18.0.0)